compiler/passes/normalize.cpp

```diff
@@ -3350,12 +3371,14 @@ static void expandQueryForGenericTypeSpecifier(FnSymbol*  fn,
 }
 
 static TypeSymbol* getTypeForSpecialConstructor(CallExpr* call) {
-  if (call->isNamed("_build_tuple")) {
+  if (call->isNamed("_build_tuple") || call->isNamed("*")) {
```

Comment:
Might be good to have an assert that the call is not PRIM_MULT (in case we change the behavior of "isNamed" later).